pkg/requests/requests.go

```diff
@@ -0,0 +1,155 @@
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const (
+	apiRepositories = "_apis/git/repositories"
+	apiProjects     = "_apis/projects"
+	itemsPath       = "items"
+	refsPath        = "refs"
+)
+
+// GetAndParseJSON invokes a HTTP request with basic auth.
+// On success the response body will be unmarshalled as JSON.
+func GetAndParseJSON(result interface{}, user, token string, urlPath *url.URL) error {
+	body, err := getBodyFromRequest(user, token, urlPath)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(body, &result)
+
+	return err
+}
```

Comment:
Suggest naming this `GetUnmarshalJSON` or something similar, replacing `Parse` with `Unmarshal` as it's the nomenclature in Go to show that you expect a pointer that will be populated via reflection